### PR TITLE
SftpAdapter: fix issue with `$connection->rawlist()` returning false

### LINKED ISSUE #1300: `Invalid argument supplied for foreach()` in SftpAdapter when calling `listContents`
### Bug Report

|    Q        |   A
|------------ | ------
| BC Break    | -
| Version     | 2.0.5

#### Summary

I have an issue when I want to list directory contents with a path that does not exist on the server when using `SftpAdapter`.

I get an error `Invalid argument supplied for foreach()`.

#### How to reproduce

This call returns `false`: https://github.com/thephpleague/flysystem/blob/2.x/src/PhpseclibV2/SftpAdapter.php#L258

Which can happen in some cases: https://github.com/phpseclib/phpseclib/blob/master/phpseclib/Net/SFTP.php#L849

### PR DESCRIPTION
**Pull request by dmaicher**
Fixes https://github.com/thephpleague/flysystem/issues/1300

### CODE REVIEW

**Comment by dmaicher** [comment]
This github action approval thing is super annoying... can someone approve it so I can see if tests are passing or failing? :confused:

**Comment by frankdejonge** [comment]
@dmaicher they are running

File: src/PhpseclibV2/SftpAdapterTest.php [hunk 1]
```diff
@@ -194,6 +194,15 @@ public function failing_to_chmod_when_writing_with_a_stream(): void
         }
     }
 
+    /**
+     * @test
+     */
+    public function list_contents_directory_does_not_exist(): void
```
Comment (dmaicher) [inline src/PhpseclibV2/SftpAdapterTest.php:200]: Ok so this test reproduced my issue

```
1) League\Flysystem\PhpseclibV2\SftpAdapterTest::list_contents_directory_does_not_exist
Invalid argument supplied for foreach()

/home/runner/work/flysystem/flysystem/src/PhpseclibV2/SftpAdapter.php:264
/home/runner/work/flysystem/flysystem/src/PhpseclibV2/SftpAdapterTest.php:203

**Comment by dmaicher** [comment]
@frankdejonge could you approve the actions one more time please?

**Comment by frankdejonge** [comment]
@dmaicher done

**Comment by dmaicher** [comment]
Build fail seems unrelated? Ready for review then

**Comment by frankdejonge** [comment]
Cheers, @dmaicher. I'll be tagging a release shortly.

**Comment by frankdejonge** [comment]
This is released as `2.0.8`, thank you for your contribution @dmaicher 👍

**Pull request merged**